.github/workflows/release.yml

```diff
@@ -0,0 +1,142 @@
+
+    ```
+    pecl install mongodb-{0}
+    ```
+
+    or update with:
+
+    ```
+    pecl upgrade mongodb-{0}
+    ```
+
+    Windows binaries are attached to the GitHub release notes.
+
+jobs:
+  prepare-release:
+    name: "Prepare release"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Create release output"
+        run: echo '🎬 Release process for version ${{ inputs.version }} started by @${{ github.triggering_actor }}' >> $GITHUB_STEP_SUMMARY
+
+      - uses: actions/checkout@v4
+        with:
+          # fetch-depth 0 is required to fetch all branches, not just the branch being built
+          fetch-depth: 0
+          submodules: true
+          token: ${{ env.GH_TOKEN }}
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php-version }}"
+          tools: "phpize"
+
+      - name: "Update version information to stable release"
+        run: ./bin/update-release-version.php stable
+
+      - name: "Read current package version"
+        run: echo "PACKAGE_VERSION=$(./bin/update-release-version.php version)" >> "$GITHUB_ENV"
+
+      # Sanity check - the version from the input and the one determined by phongo_version.h need to be the same
+      - name: "Check version for consistency"
+        if: ${{ inputs.version != env.PACKAGE_VERSION }}
+        # We exit with an error to abort the workflow. This is only run if the versions don't match
+        run: |
+          echo '❌ Release failed due to version mismatch: expected ${{ inputs.version }}, got ${{ env.PACKAGE_VERSION }} from code' >> $GITHUB_STEP_SUMMARY
+          exit 1
+
+      #
+      # Preliminary checks done - commence the release process
+      #
+
+      - name: "Set git author information"
+        run: |
+          git config user.name "${GIT_AUTHOR_NAME}"
+          git config user.email "${GIT_AUTHOR_EMAIL}"
+
+      # Create the "Package x.y.z" commit that will be the base of our tag
+      - name: "Create release commit"
+        run: git commit -m "Package ${{ env.PACKAGE_VERSION }}" phongo_version.h
+
+      # Create a draft release with a changelog
+      # TODO: Consider using the API to generate changelog
+      - name: "Create draft release with generated changelog"
+        run: gh release create ${{ env.PACKAGE_VERSION }} --target ${{ github.ref_name }} --generate-notes --draft
+
+      - name: "Read changelog from draft release"
+        run: gh release view ${{ env.PACKAGE_VERSION }} --json body --template '{{ .body }}' >> changelog
+
+      # TODO: Sign tag
+      - name: "Create release tag"
+        run: git tag -a -F changelog ${{ env.PACKAGE_VERSION }}
+
+      - name: "Update version information to next patch development release"
+        run: |
+          ./bin/update-release-version.php patch
+          git commit -m "Back to -dev" phongo_version.h
+
+      # TODO: Manually merge using ours strategy. This avoids merge-up pull requests being created
```

Comment:
I'll defer this to a separate PR as it requires some work to make this logic from the merge-up action reusable here.

Reply:
Reminder to ticket this if needed so we don't lose track of it.

I assume we already have a ticket to track the earlier TODO for release signing.

Reply:
PHPC-2375